### PR TITLE
feat(attribute-completeness): improve attribute-completeness for single attribute

### PR DESCRIPTION
**Pull request by Gigaszi**
### Corresponding issue
Closes #
#805
### New or changed dependencies
-

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-api/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-api/blob/main/CHANGELOG.md)

### TODO

- [x] Let's put attribute metadata information also into `/metadata` as with the other metadata.
- [x] Fix figure (needle and figure heading)
- [x] Make attribute request parameter optional for other indicators. This request parameter should be required for the Attribute Completeness
- [x] Take a look at swagger → Swagger should not have Attribute Completeness indicator as example. Example should not container a `attribute` key.
- [x] Add Attribute Enum and use it pydantic request models to generate a clean OpenAPI spec
- [x] Add validation of topic-attribute combination to attribute-completeness

The UI-related tasks were moved to an issue in the Ohsome Dashboard project:

https://github.com/GIScience/ohsome-dashboard/issues/42

Some forward-looking backend-related tasks were moved to a new issue:

#819

### CODE REVIEW

**Comment by matthiasschaub** [comment]
- [ ] Let's put attribute metadata information also into `/metadata` as with the other metadata.

File: ohsome_quality_api/utils/validators.py [hunk 1]
```diff
@@ -14,6 +20,17 @@
 from ohsome_quality_api.utils.helper_geo import calculate_area
 
 
+def validate_attribute_topic_combination(attribute: AttributeEnum, topic: TopicEnum):
+    """As attributes are only meaningful for a certain topic,
+    we need to check if the given combination is valid."""
+
+    valid_attributes_for_topic = get_attributes()[topic]
+    all_attributes = [attribute for attribute in valid_attributes_for_topic]
```
Comment (mmerdes) [inline ohsome_quality_api/utils/validators.py:28]: the name all_attributes is misleading - use `valid_attributes_names` instead?

File: tests/integrationtests/api/test_indicators.py [hunk 2]
```diff
@@ -120,6 +121,68 @@ def test_indicators(
...
+    headers,
+    schema,
+):
+    endpoint = ENDPOINT + "attribute-completeness"
+    parameters = {"bpolys": bpolys, "topic": "building-count", "attribute": "height"}
+    response = client.post(endpoint, json=parameters, headers=headers)
+    assert schema.is_valid(response.json())
+
+
+@oqapi_vcr.use_cassette
```
Comment (mmerdes) [inline tests/integrationtests/api/test_indicators.py:137]: Can we drop the cassette here as there is no remote access to ohsome API?

File: tests/integrationtests/api/test_indicators.py [hunk 3]
```diff
@@ -165,6 +228,25 @@ def test_minimal_include_figure_false(client, bpolys, headers, schema):
...
+def test_minimal_additional_parameter_foo(client, bpolys, headers, schema):
+    endpoint = ENDPOINT + "minimal"
+    parameters = {"bpolys": bpolys, "topic": "minimal", "attribute": "foo"}
+    response = client.post(endpoint, json=parameters, headers=headers)
+    assert response.status_code == 422
+    content = response.json()
+    assert content["type"] == "RequestValidationError"
+
+
+@oqapi_vcr.use_cassette
```
Comment (mmerdes) [inline tests/integrationtests/api/test_indicators.py:240]: see above

File: tests/integrationtests/api/test_ohsome_timeout.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
 from tests.integrationtests.utils import AsyncMock
 
 
+# TODO: could/should this be converted to a parameterized test?
```
Comment (mmerdes) [inline tests/integrationtests/api/test_ohsome_timeout.py:19]: will be changed anyway when the report endpoint is removed

**Review by mmerdes — CHANGES_REQUESTED** [review]
review jointly done by @gigaszi, @sfendrich, and @mmerdes

File: ohsome_quality_api/utils/validators.py (hunk 1 above)
Comment (mmerdes) [reply ohsome_quality_api/utils/validators.py:28]: fixed

File: tests/integrationtests/api/test_indicators.py (hunk 3 above)
Comment (mmerdes) [reply tests/integrationtests/api/test_indicators.py:240]: fixed

**Review by mmerdes — APPROVED** [review]
ok

**Pull request merged**